Currently working on TeamStandingStatsCalculation--IN PROGRESS

Circador opened the pull request:
TODO:
- Finish TeamStatsDTO class that contains all of the relevant information to be displayed on team standings page.
- Finish TeamStandingsStatsCalculator by going through the statsGenerationRequest's MatchDTO's and adding those results into an array of TeamStatsDTO
- Work on the same model for the FullIndividualStatsCalculator

Circador committed:
Currently working on TeamStandingStatsCalculation.
TODO:
- Finish TeamStatsDTO class that contains all of the relevant information to be displayed on team standings page.
- Finish TeamStandingsStatsCalculator by going through the statsGenerationRequest's MatchDTO's and adding those results into an array of TeamStatsDTO
- Work on the same model for the FullIndividualStatsCalculator

Circador committed:
Still working on MatchTeamStatsDTO and cleaned up TeamStandingsStatsCalculator calculation structure

Circador committed:
Daily Commit 01/24/2018

Added new Calculators to match the neg5 online interface and new ResultDTOs for each of the calculators.
TODO:
- finish implementing TeamStandings, FullIndividual, IndividualStandings, RoundReport Calculators and DTOs

mostafab commented on app/calculators/FullTeamStatsCalculator.java:
```diff
@@ -0,0 +1,71 @@
+            int numTeams = curMatch.getTeams().size();
+            // add and calculate results for each team of the match
+            for(int j = 0; j < numTeams; j++){
+                MatchTeamDTO curTeam = curMatch.getTeams().get(j);
```
For consistency's sake, let's make this inner loop look like the outer for-loop that's using an iterator.

mostafab commented on app/calculators/FullTeamStatsCalculator.java:
```diff
@@ -0,0 +1,71 @@
+package calculators;
+
+import interfaces.*;
```
We shouldn't use the wildcard import, it's better to import each needed class individually. You should be able to configure this in whatever IDE you're using

mostafab reviewed (commented):
It'd be great to include unit tests for the calculation methods

mostafab commented on app/calculators/TeamStandingsStatsCalculator.java:
```diff
@@ -1,15 +1,76 @@
 package calculators;
 
-import interfaces.StatsGenerationRequestDTO;
+import interfaces.*;
```
Same comment about wildcard imports

mostafab commented on app/calculators/TeamStandingsStatsCalculator.java:
```diff
@@ -1,15 +1,76 @@
+import interfaces.*;
 import interfaces.stats.TeamStandingsStatsCalculationResultDTO;
 
+import java.util.*;
```
Same as above

mostafab commented on app/calculators/TeamStandingsStatsCalculator.java:
```diff
@@ -1,15 +1,76 @@
+     */
     public TeamStandingsStatsCalculationResultDTO calculate(StatsGenerationRequestDTO statsGenerationRequest) {
-        return new TeamStandingsStatsCalculationResultDTO();
+		TeamStandingsStatsCalculationResultDTO results = new TeamStandingsStatsCalculationResultDTO();
```
We should have consistent spacing in files. I'll add a checkstyle rule in the future

mostafab commented on app/interfaces/MatchTeamStatsDTO.java:
```diff
@@ -0,0 +1,167 @@
+    private Integer pointsAgainst;
+    private List<TeamTossupValuesDTO> tossupValues;
+    private Integer TUH;
+    private Float PPTH;
```
We should be using Doubles instead of Floats for decimal values.

mostafab commented on app/interfaces/MatchTeamStatsDTO.java:
```diff
@@ -0,0 +1,167 @@
+    public static final DecimalFormat DF = new DecimalFormat( "#.##" );
+
+    // Display stats
+    private String opponent;
```
Some other fields that should be included in this DTO are: opponentId, round, overtime tossups, the match id

mostafab commented on app/interfaces/MatchTeamStatsDTO.java:
```diff
@@ -0,0 +1,167 @@
+    private Integer pointsTotal;
+    private Integer pointsAgainst;
+    private List<TeamTossupValuesDTO> tossupValues;
+    private Integer TUH;
```
We should follow camel-case naming conventions, so this would be 'tuh`, PPB would be 'ppb', etc.

mostafab commented on app/interfaces/MatchTeamStatsDTO.java:
```diff
@@ -0,0 +1,167 @@
+        tossupValues = new ArrayList<>();
+
+        // initialize opponent info
+        List<MatchTeamDTO> teams = matchDTO.getTeams();
```
Can we move this out to a private method? It'll be cleaner than doing all the code inside the constructor

mostafab commented on app/interfaces/MatchTeamStatsDTO.java:
```diff
@@ -0,0 +1,167 @@
+
+    // Display stats
+    private String opponent;
+    private String result;
```
Let's make the result be an enum ("W", "L", "T"). It'll be cleaner that way

mostafab commented on app/interfaces/TeamStandingStatsDTO.java:
```diff
@@ -0,0 +1,69 @@
+    private float winPct;
+    private float PPG;
+    private float PAPG;
+    private int pow;
```
We should name these "numPowers", "numBase", "numNegs" for clarity. This DTO should also have an array of tossup values similar to MatchTeamStatsDTO, but as a sum instead of match specific

mostafab commented on app/interfaces/TeamStandingStatsDTO.java:
```diff
@@ -0,0 +1,69 @@
+    private int neg;
+    private float margin;
+    private int TUH;
+    private float PPTH;
```
Doubles

mostafab commented on app/calculators/IndividualStandingsStatsCalculator.java:
```diff
@@ -0,0 +1,4 @@
+package calculators;
+
+public class IndividualStandingsStatsCalculator {
```
Small oversight, this should implement StatsCalculator

mostafab commented on app/calculators/RoundReportStatsCalculator.java:
```diff
@@ -0,0 +1,4 @@
+package calculators;
+
+public class RoundReportStatsCalculator {
```
Should also implement StatsCalculator

mostafab commented on app/interfaces/MatchTeamStatsDTO.java:
```diff
@@ -0,0 +1,167 @@
+ */
+public class MatchTeamStatsDTO extends MatchTeamDTO{
+    // formatter
+    public static final DecimalFormat DF = new DecimalFormat( "#.##" );
```
Will this be used somewhere else? If not, it should be private

mostafab commented on app/interfaces/TeamTossupValuesDTO.java:
```diff
@@ -0,0 +1,42 @@
+
+public class TeamTossupValuesDTO {
+
+    private Integer value;
```
Can we add the a field of type `TossupType` to this?

Circador committed:
Daily Commit 01/29/2018

Finished first implementation of TeamStandingStatsCalculator based on the TeamStandingStatsDTO
TODO:
- test TeamStandingStatsCalculator
- finish FullTeamStatsCalculator based on MatchTeamStatsDTO
- continue writing documentation

Circador committed:
Daily Commit 01/30/2018

Finished first implementation of IndividualStandingStatsCalculator based on the IndividualPlayerDTO
TODO:
- test TeamStandingStatsCalculator & IndividualStandingStatsCalculator
- finish FullTeamStatsCalculator based on MatchTeamStatsDTO
- finish FullIndividualStatsCalculator based on FullIndividualPlayerDTO
- continue writing documentation

Commit:
created code of conduct

Circador committed:
Merge pull request #1 from mren36/master

created code of conduct

Circador committed:
Reorganizing DTOs

Circador committed:
Added issues to fix.

Circador committed:
Daily Commit 2/27/18 (and the rest of February)

- finished implementing all four stats calculators
- organized DTO subfolder structure
- added more DTOs
- added contributing guidelines
